prototyper/prototyper/src/platform/console.rs

```diff
@@ -127,3 +132,80 @@ impl ConsoleDevice for UartBflbWrap {
         count
     }
 }
+
+/// PL011 UART wrapper for RustSBI console
+pub struct UartPl011Wrap {
+    uart: UnsafeCell<Uart<'static>>,
+}
+
+impl UartPl011Wrap {
+    /// Create a new PL011 UART wrapper
+    pub fn new(base: usize) -> Self {
```

Comment:
This function effectively dereferences any raw address thus should be considered unsafe. I'll handle this later in following PRs.